plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/AnalysisKickOff.java

```diff
@@ -89,6 +90,13 @@ public boolean setUp(final IJavaElement iJavaElement) {
 	public void run() {
 		if(this.curProj == null)
 			return;
+		String javaVersion = System.getProperty("java.version");
+		JavaVersion systemJavaVersion = new JavaVersion(javaVersion);
+		JavaVersion requiredJavaVersion = new JavaVersion(Constants.CC_JAVA_VERSION);
+		if (javaVersion != "" && systemJavaVersion.compareTo(requiredJavaVersion) == 1) {

```

Comment:
`javaVersion != ""` is always false. In Java Strings are immutable and both sides are always to unique objects.
Do you need the check at all? What happens if the version is empty? Please raise a warning "Was not able to detect the Java version" and cancel the analysis in that case as well.